src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseSign1Message.cs

```diff
@@ -85,36 +130,75 @@ public static byte[] Sign(byte[] content!!, AsymmetricAlgorithm key!!, HashAlgor
                 writer.WriteByteString(content);
             }
 
-            writer.WriteByteString(signature);
+            int expectedToBeSignedSize = ComputeToBeSignedEncodedSize(SigStructureCoxtextSign1, encodedProtectedHeaders, content);
+
+            Span<byte> toBeSignedBuffer = buffer;
+            byte[]? rentedToBeSignedBuffer = null;
+            int signatureBytesWritten;
+
+            // It is possible for toBeSigned to be bigger than the COSE message length that we used to determine the size of our buffer.
+            // we rent a bigger buffer if that's the case.
+            if (buffer.Length < expectedToBeSignedSize)
+            {
+                rentedToBeSignedBuffer = ArrayPool<byte>.Shared.Rent(expectedToBeSignedSize);
```

Comment:
I'd like to see this rent go away and us be able to create the TBS with non-contiguous memory (and hashed using IncrementalHash).

But I expect that'll be forced upon us when we add support for Stream-based detached content; so it's OK for now.